src/modules/accommodations/dto/request/search-accommodation-params.dto.ts

```diff
@@ -0,0 +1,30 @@
+import { IsDate, IsOptional, IsString, Min, IsInt, Max } from 'class-validator';
+
+export class SearchAccommodationQueryParamsDto {
+  @IsString()
+  @IsOptional()
+  search: string;
+
+  @IsString()
+  @IsOptional()
+  category: string;
+
+  @IsDate()
+  @IsOptional()
+  fromDate: Date;
+
+  @IsDate()
+  @IsOptional()
+  toDate: Date;
+
+  @IsInt()
+  @IsOptional()
+  @Min(1)
+  page: number = 1;
```

Comment:
first you need transform this into number then you can apply IsInt and other decorators. You can use the decorator that I created for this purpose